frontend/src/components/InstanceForm/common.ts

```diff
@@ -126,6 +163,23 @@ export const calcDataSourceUpdateMask = (
   if (updateAuthenticationPrivateKey) {
     updateMask.add("authentication_private_key");
   }
+  
+  // Always add extra_connection_parameters to update mask
+  // This is needed even if they're empty or haven't changed, to ensure proper handling of parameters
+  updateMask.add("extra_connection_parameters");
+  
+  // Make sure editing has the correct extraConnectionParameters
+  if (editState.extraConnectionParameters) {
+    // Clone the map manually to ensure it's a plain object, not a Proxy
+    const params: Record<string, string> = {};
+    Object.entries(editState.extraConnectionParameters).forEach(([key, value]) => {
+      params[key] = value;
+    });
+    
+    editing.extraConnectionParameters = params;
+  } else {
+    editing.extraConnectionParameters = {}; 
```

Comment:
Please NOT update the input params in a function named `calcDataSourceUpdateMask`